sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryUtils.java

```diff
@@ -82,11 +85,18 @@
 
                 if (!shouldRetryResult.shouldRetry) {
                     retryPolicy.updateEndTime();
-                    if(shouldRetryResult.exception == null) {
-                        return Mono.error(e);
-                    } else {
-                        return Mono.error(shouldRetryResult.exception);
+
+                    final Throwable errorToReturn = shouldRetryResult.exception != null ? shouldRetryResult.exception : e;
+                    final Mono<T> failure = Mono.error(errorToReturn);
+
+                    if (shouldRetryResult.policyArg != null &&
```

Comment:
Can you please add a comment here what `policyArg.getValue0()` is ?

Reply:
Fixed in next iteration